stdlib/public/Concurrency/CMakeLists.txt

```diff
@@ -260,8 +260,15 @@ if(SWIFT_SHOULD_BUILD_EMBEDDED_STDLIB AND SWIFT_SHOULD_BUILD_EMBEDDED_CONCURRENC
       if("${mod}" MATCHES "armv6m")
         continue()
       endif()
-      set(extra_c_compile_flags -D__MACH__ -D__APPLE__ -ffreestanding -stdlib=libc++)
-      set(extra_swift_compile_flags -Xcc -D__MACH__ -Xcc -D__APPLE__ -Xcc -ffreestanding)
+
+      if("${mod}" MATCHES "-macos$")
+        set(extra_c_compile_flags -D__MACH__ -D__APPLE__ -ffreestanding -stdlib=libc++)
+        set(extra_swift_compile_flags -Xcc -D__MACH__ -Xcc -D__APPLE__ -Xcc -ffreestanding)
```

Comment:
Why are `-D__MACH__ -D__APPLE__` needed? those should be present with the `apple` vendor.

Reply:
That's unclear to me. Keeping as is, changing it doesn't seem relevant to this PR.

Reply:
These are needed for <arch>-apple-none-macho but shouldn't be needed otherwise. the `__MACH__` one should only be needed for os ~= darwin but sadly `__MACH__` is abused to also mean `__MACHO__`